webhook_to_fedora_messaging/config/__init__.py

```diff
@@ -2,19 +2,47 @@
 #
 # SPDX-License-Identifier: GPL-3.0-or-later
 
-from os import environ
 
-import tomllib
+import logging
+import os
+from logging.config import dictConfig
+from pathlib import Path
 
-from webhook_to_fedora_messaging.exceptions import ConfigError
+from sqlalchemy_helpers.fastapi import AsyncDatabaseManager
 
+from webhook_to_fedora_messaging import database
+from webhook_to_fedora_messaging.config import standard
 
-def get_config() -> dict:
-    path = environ["W2FM_APPCONFIG"]
-    try:
-        with open(path, "rb") as file:
-            return tomllib.load(file)["flaskapp"]
-    except FileNotFoundError as expt:
-        raise ConfigError(f"Configuration file '{path}' was not found") from expt
-    except KeyError as expt:
-        raise ConfigError(f"Configuration key '{expt}' could not be read") from expt
+
+logger = logging.getLogger(__name__)
+
+
+def setup_config() -> dict:
+    path = os.getenv("W2FM_CONFIG")
+    confdict = {}
+
+    with open(path) as file:
+        exec(compile(file.read(), path, "exec"), confdict)  # noqa : S102
+
+    confkeys = (
+        "database_url",
+        "service_host",
+        "service_port",
+        "reload_on_change",
+        "main_logger_conf",
+        "wsgi_logger_conf",
+        "fasjson_url",
+    )
+    for conf in confkeys:
+        if conf not in confdict:
+            continue
+        setattr(standard, conf, confdict[conf])
+    dictConfig(standard.main_logger_conf)
+    logger.info("Reading the configuration again.")
+
+
+def setup_database_manager() -> None:
```

Comment:
I would prefer having this function in the `database.py` module, so that a module's global variable is not changed from outside (it's harder to track).

Reply:
To do that, I would have to use `global` to make a lasting contextual change to the `db` object in the `database` module. Is there a way in which I can achieve what you are asking me to do without the use of `global`?

Reply:
Well, it is a module-level variable, stored at module-level. Changing it from another module does not make it non-global. But I think it makes it worse: it's already bad enough that this variable is read from anywhere, we should try to avoid writing to it from anywhere.

To avoid accessing it as a global variable, we could do something like:

```py
_db = None
def setup_database_manager()
    global _db
    _db = AsyncDatabaseManager(...)

def get_database_manager():
    return _db
```

It's a bit more boilerplate but it makes for a "cleaner" module API I think.

In Flask there is the concept of global objects, such as `g` and `current_app`, but FastAPI doesn't have those, or any kind of global registry, unless I missed it.